modular_skyrat/modules/modular_implants/code/nifsofts/soulcatcher.dm

```diff
@@ -0,0 +1,86 @@
+	purchase_price = 150 //RP tool
+	persistence = TRUE
+	/// What is the linked soulcatcher datum used by this NIFSoft?
+	var/datum/component/soulcatcher/linked_soulcatcher
+	/// What action to bring up the soulcatcher is linked with this NIFSoft?
+	var/datum/action/innate/soulcatcher/soulcatcher_action
+	/// a list containing saved soulcatcher rooms
+	var/list/saved_soulcatcher_rooms = list()
+
+/datum/nifsoft/soulcatcher/New()
+	. = ..()
+	soulcatcher_action = new
+	soulcatcher_action.Grant(linked_mob)
+	soulcatcher_action.parent_nifsoft = WEAKREF(src)
+
+	var/obj/item/organ/internal/cyberimp/brain/nif/target_nif = parent_nif
+	linked_soulcatcher = target_nif.AddComponent(/datum/component/soulcatcher)
```

Comment:
this shouldn't be a hard ref to the component, and even then, this should also have a parent qdeleting signal to null it if the component is deleted

Reply:
This is a separate issue with NIFs, if you would like, I could make a separate PR to fix this.

Reply:
wait nevermind, I was thinking of something else.